onmt/inputters/dynamic_iterator.py

```diff
@@ -157,54 +170,111 @@ def _init_datasets(self):
             self.corpora, self.transforms, self.corpora_info,
             skip_empty_level=self.skip_empty_level,
             stride=self.stride, offset=self.offset)
-        self.dataset_adapter = DatasetAdapter(self.fields, self.is_train)
         datasets_weights = {
             ds_name: int(self.corpora_info[ds_name]['weight'])
             for ds_name in datasets_iterables.keys()
         }
-        if self.is_train:
+        if self.task == CorpusTask.TRAIN:
             self.mixer = WeightedMixer(datasets_iterables, datasets_weights)
         else:
             self.mixer = SequentialMixer(datasets_iterables, datasets_weights)
         self.init_iterators = True
 
+    def _adapt_bucket(self, tuple_bucket):
```

Comment:
A docstring wouldn't hurt here, especially since it looks like some patch related to copy mechanism.